cli/src/commands/metadata.rs

```diff
@@ -39,9 +40,11 @@ pub async fn run(opts: Opts) -> color_eyre::Result<()> {
             }
         };
 
-        retain_metadata_pallets(metadata_v14, |pallet_name| {
+        retain_metadata_pallets(&mut metadata_v15, |pallet_name| {
```

Comment:
Ah so to retain specific pallets maybe it makes sense then to update the metadata to V15 so that we don't have to have logic to strip the metadata to certain pallets for different versions of it. But that's ok; we can make a note that if you use the `--pallets` thing then it'll update to V15 (currently unstable) as well.

Reply:
Yep, will document around the `--pallets` option that the metadata will be updated to V15.

Another option for handling this would be to have bidirectional `From` traits.
We currently implement V14 -> V15 and we could implement V15 -> V14 to not affect the shape of the metadata while keeping the code footprint in `metadata/retain.rs` as small as possible.
However, that adds extra code complexity since upgrading and downgrading the metadata requires a fairly large method.

Weighing the options, I think that documenting this CLI option provides a better outcome for us